### ISSUE TITLE
Contact labels (groups) get depopulated without warning

### ISSUE THREAD

**Issue by LearningAsIGo71** [SimpleMobileTools/Simple-Contacts#867]
I am running the latest version of GrapheneOS (GOS) Android 12 on a Pixel 5a. I used Simple Contacts to import my contacts from a Google Contacts vcf file. All 2000+ contacts imported as well as all 30+ contact group labels, with all the group labels properly populated with the corresponding contacts. So far so good!

Prior to the import I had only two contact group labels created for GrapheneOS's (GOS) contacts which identically matched two of the 30+ imported Google contact group labels. The two matching GOS contact group labels got populated with the corresponding contacts that Simple Contacts imported from the the vcf file, but for some reason Simple Contacts did not create any additional group labels in the GOS contacts app. 

Then at some point (I'm not sure how long after the initial vcf import) all of Simple Contacts' 30+ contact group labels became completely empty (i.e. no longer contain any contacts) with the exception of the two contact group labels that match the two GOS contact group labels! These two group labels are still identically populated in both the GOS contacts app and in Simple Contacts. 

It seems what is happening is that Simple Contacts does a perfect job of importing from Google Contacts vcf file (including contact group labels), but the group labels do not get imported into the GrapheneOS contacts app, and so when Simple Contacts syncs with the GOS contacts app, whatever group labels do not exist in the GOS contacts app get completely depopulated on the Simple Contacts side. 

Perhaps one workaround would be to manually create all 30+ group labels in the GOS contacts app prior to the Simple Contacts import of the Google vcf file, then at the time of import all Simple Contact's group labels would get populated with the corresponding in contacts in GOS contacts and therefore remain in Simple Contacts when a sync occurs?

**Comment by tibbi** [comment]
what contact source are you saving the contact to? I mean the last field at the contact details screen. Does it say your email, Phone, SIM or something like that?

**Comment by LearningAsIGo71** [comment]
The last field in the contacts details screen says "Addresses". For certain contacts "Signal" is also listed after Addresses, referring to Signal Messenger.

**Comment by LearningAsIGo71** [comment]
Okay, I just realized "Addresses" is the name of my GeneralSync account. GeneralSync syncs contacts and calendars between devices (i.e. laptops, desktops, phones) on a common WiFi network. Looking more closely it is strange. It appears there are no GOS Contacts, but just GOS contacts linked to the GeneralSync account. In fact, when I go to add a contact or group label via GOS Contacts I only have the option of adding to the GeneralSync account. I don't see any way to simply add a contact to GOS Contacts apart from the GeneralSync account.

**Comment by LearningAsIGo71** [comment]
I decided to uninstall GeneralSync and delete all the contacts in Simple Contacts. I then re-imported the Google contacts vcf file into Simple Contacts. All the contacts look to have imported properly, including all Contact Groups / Labels populated with the appropriate contacts. I then exported all the Simple contacts to a vcf and imported the vcf into GrapheneOS contacts, which come to find out apparently does not support Groups / Labels. After all the contacts were imported into GrapheneOS contacts, all of the Simple Contacts Groups lost all their associated contacts. In other words, all the Contact Groups are once again empty.

**Comment by LearningAsIGo71** [comment]
I decided to uninstall GeneralSync and delete all the contacts and groups in Simple Contacts (so grateful for the feature that allows all contacts to be selected and deleted at once!). Upon doing this all my contacts were deleted from GrapheneOS contacts as expected, with the exception of four contacts (out of over 2500), which I easily selected and deleted. All groups / labels were also deleted from GrapheneOS (GOS) and GOS contains no ability on it’s own to create groups / labels which I can find. 

I then went to re-import the Google contacts vcf file into Simple Contacts and noticed that Simple Contacts defaults “Addresses” from GeneralSync as a target location. I decided to try using it and see what happens and Simple Contacts began displaying “Error: java.lang.illegalArgumentException” one after another after another (I suppose it was going to do this for all 2500+ contacts that could no longer be imported into GeneralSync). I closed out of Simple Contacts but the error messages kept appearing. I wasn’t about to wait for all 2500+ error messages so I restarted the phone. 

Upon restart I went to re-import the Google contacts into Simple Contacts and it still defaults to “Addresses” as a target location. Upon tapping it I then was able to select “Phone storage (not visible by other apps).” All the contacts look to have imported properly, including all Contact Groups / Labels populated with the appropriate contacts. 

For whatever reason GOS contacts is unable to import the Google contacts vcf file which Simple Contacts does with ease. GOS contacts simply gives the error message “Couldn’t import VCard.”

So I used Simple Contacts to export all the 2500+ Simple contacts to a vcf and was able to import this vcf into GrapheneOS contacts. But after all the contacts were imported into GOS Contacts, all of the Simple Contacts Groups once again lost all their associated contacts! In other words, all the 30+ Contact Groups remain in Simple Contacts, but all of groups are void of any contacts. It’s like Simple Contacts sees GOS Contacts has no group labels and decides likewise none of its contacts should be associated with any labels. 

Ironically however, after importing the Simple Contacts vcf file into GOS Contacts, although none of the group/labels were imported into GOS Contacts, GOS Contacts now has the ability to make Labels via Menu > Create label, although none of the group labels were imported!

**Comment by LearningAsIGo71** [comment]
Well this situation looks like it is not going to get addressed so I gave EteSync a try and all my contacts from the Google vcf file got imported with all groups/labels in tact and populated and accessible from phone's stock Contacts app! I then imported EteSync contacts into GeneralSync and now have a seamless way to sync my contacts between phones on the same LAN as well as with contacts on PCs running Thunderbird or Outlook. Now if I ever lose or break a phone I will be able to simply install GeneralSync, connect to the LAN and restore all my phone's contacts. Almost as easy as using Google! LOL
